Mathlib/Algebra/Algebra/Subalgebra/CenterAndCentralizer.lean

```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import Mathlib.Algebra.Algebra.Subalgebra.Basic
+import Mathlib.RingTheory.TensorProduct.Basic
+import Mathlib.RingTheory.Adjoin.Basic
+import Mathlib.LinearAlgebra.TensorProduct.Basis
+import Mathlib.LinearAlgebra.FreeModule.Basic
+
+/-!
+# Properties of centers and centralizers
+
+This file contains theorems about the center and centralizer of a subalgebra.
+
+## Main results
+
+Let `R` be a commutative ring and `A` and `B` two `R`-algebras.
+- `Subalgebra.centralizer_sup`: if `S` and `T` are subalgebras of `A`, then the centralizer of
+  `S ⊔ T` is the intersection of the centralizer of `S` and the centralizer of `T`.
+- `Subalgebra.centralizer_range_includeLeft_eq_center_tensorProduct`: if `B` is free as a module,
+  then the centralizer of `A` in `A ⊗ B` is `C(A) ⊗ B` where `C(A)` is the center of `A`.
+- `Subalgebra.centralizer_range_includeRight_eq_center_tensorProduct`: if `A` is free as a module,
+  then the centralizer of `B` in `A ⊗ B` is `A ⊗ C(B)` where `C(B)` is the center of `B`.
+-/
+
+namespace Subalgebra
+
+section CommSemiring
+
+variable {R : Type*} [CommSemiring R]
+variable {A : Type*} [Semiring A] [Algebra R A]
+
+lemma centralizer_sup (S T : Subalgebra R A) :
+    centralizer R ((S ⊔ T : Subalgebra R A) : Set A) = centralizer R S ⊓ centralizer R T := by
```

Comment:
Can you provide the `iSup` and `sSup` versions as well?

Reply:
I'd only request the `iSup` version as it's easy to use [sSup_eq_iSup'](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/CompleteLattice.html#sSup_eq_iSup') when you do encounter `sSup`.

Reply:
I have added iSup